dbft: run new node in miner mode encounter error and stop with error "panic: runtime error: index out of range [-1]"

#### System information

Geth Version: 0.3.0-unstable
Git Commit: 05a48b0ff50c858c8830183051aafb22f3f5515e
Git Commit Date: 20241107
Architecture: amd64
Go Version: go1.22.2
Operating System: linux

#### Expected behaviour
start new node in miner mode  should not stop and run well

#### Actual behaviour
node stop with error when start at the very beginning
```

INFO [11-27|11:41:16.281] Enabling metrics collection
INFO [11-27|11:41:16.285] Maximum peer count                       ETH=50 total=50
INFO [11-27|11:41:16.293] Smartcard socket not found, disabling    err="stat /run/pcscd/pcscd.comm: no such file or directory"
INFO [11-27|11:41:16.313] Set global gas cap                       cap=50,000,000
INFO [11-27|11:41:16.313] Initializing the KZG library             backend=gokzg
INFO [11-27|11:41:16.755] Allocated trie memory caches             clean=154.00MiB dirty=256.00MiB
INFO [11-27|11:41:16.763] Using pebble as the backing database
INFO [11-27|11:41:16.763] Allocated cache and file handles         database=/home/bruce/new/main/go-ethereum/privnet/seven/node9/geth/chaindata cache=512.00MiB handles=524,288
INFO [11-27|11:41:16.920] Opened ancient database                  database=/home/bruce/new/main/go-ethereum/privnet/seven/node9/geth/chaindata/ancient/chain readonly=false
INFO [11-27|11:41:16.921] State scheme set to already existing     scheme=hash
INFO [11-27|11:41:16.923] Initialising Ethereum protocol           network=2,312,251,829 dbversion=<nil>
INFO [11-27|11:41:16.931] 
INFO [11-27|11:41:16.931] ---------------------------------------------------------------------------------------------------------------------------------------------------------
INFO [11-27|11:41:16.931] Chain ID:  2312251829 (unknown)
INFO [11-27|11:41:16.931] Consensus: dBFT (proof-of-authority)
INFO [11-27|11:41:16.931] 
INFO [11-27|11:41:16.931] Pre-Merge hard forks (block based):
INFO [11-27|11:41:16.931]  - Homestead:                   #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/homestead.md)
INFO [11-27|11:41:16.931]  - Tangerine Whistle (EIP 150): #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/tangerine-whistle.md)
INFO [11-27|11:41:16.931]  - Spurious Dragon/1 (EIP 155): #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/spurious-dragon.md)
INFO [11-27|11:41:16.936]  - Spurious Dragon/2 (EIP 158): #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/spurious-dragon.md)
INFO [11-27|11:41:16.936]  - Byzantium:                   #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/byzantium.md)
INFO [11-27|11:41:16.936]  - Constantinople:              #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/constantinople.md)
INFO [11-27|11:41:16.936]  - Petersburg:                  #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/petersburg.md)
INFO [11-27|11:41:16.936]  - Istanbul:                    #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/istanbul.md)
INFO [11-27|11:41:16.936]  - Muir Glacier:                #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/muir-glacier.md)
INFO [11-27|11:41:16.936]  - Berlin:                      #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/berlin.md)
INFO [11-27|11:41:16.936]  - London:                      #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/london.md)
INFO [11-27|11:41:16.936]  - Arrow Glacier:               #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/arrow-glacier.md)
INFO [11-27|11:41:16.936]  - Gray Glacier:                #0        (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/gray-glacier.md)
INFO [11-27|11:41:16.936] 
INFO [11-27|11:41:16.936] The Merge is not yet available for this network!
INFO [11-27|11:41:16.936]  - Hard-fork specification: https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/paris.md
INFO [11-27|11:41:16.936] 
INFO [11-27|11:41:16.936] Post-Merge hard forks (timestamp based):
INFO [11-27|11:41:16.936]  - Shanghai:                    @0          (https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/shanghai.md)
INFO [11-27|11:41:16.936]  - NeoXAMEV:                    #5       
INFO [11-27|11:41:16.936] 
INFO [11-27|11:41:16.936] ---------------------------------------------------------------------------------------------------------------------------------------------------------
INFO [11-27|11:41:16.936] 
INFO [11-27|11:41:16.940] Loaded most recent local block           number=0 hash=0926ce..ebed8c td=1 age=55y8mo2w
WARN [11-27|11:41:16.945] Failed to load snapshot                  err="missing or corrupted snapshot"
INFO [11-27|11:41:16.963] Rebuilding state snapshot
INFO [11-27|11:41:16.974] Initialized transaction indexer          range="last 2350000 blocks"
INFO [11-27|11:41:16.975] Resuming state snapshot generation       root=7b7968..0430e7 accounts=0 slots=0 storage=0.00B dangling=0 elapsed=11.909ms
INFO [11-27|11:41:16.995] Generated state snapshot                 accounts=27 slots=71 storage=7.96KiB dangling=0 elapsed=31.996ms
INFO [11-27|11:41:17.300] Enabled snap sync                        head=0 hash=0926ce..ebed8c
INFO [11-27|11:41:17.302] Gasprice oracle is ignoring threshold set threshold=2
WARN [11-27|11:41:17.319] Engine API enabled                       protocol=eth
WARN [11-27|11:41:17.332] Engine API started but chain not configured for merge yet
INFO [11-27|11:41:17.332] Starting peer-to-peer node               instance=Geth/v0.3.0-unstable-05a48b0f-20241107/linux-amd64/go1.22.2
INFO [11-27|11:41:17.388] New local node record                    seq=1,732,678,877,376 id=3197c890928dd2ea ip=127.0.0.1 udp=30314 tcp=30314
INFO [11-27|11:41:17.405] Started P2P networking                   self=enode://b00a72845bb292c0fe2a48250bcc3bff635b5e4987a48928f048b17c9c37fb5970b66158e8c201b15f8dfd1a4104e7354f8cf6df4fdea051dc1be78cdfa79916@127.0.0.1:30314
INFO [11-27|11:41:17.445] IPC endpoint opened                      url=/home/bruce/new/main/go-ethereum/privnet/seven/node9/geth.ipc
INFO [11-27|11:41:17.446] Generated JWT secret                     path=/home/bruce/new/main/go-ethereum/privnet/seven/node9/geth/jwtsecret
INFO [11-27|11:41:17.468] HTTP server started                      endpoint=[::]:8570 auth=false prefix= cors= vhosts=*
INFO [11-27|11:41:17.489] WebSocket enabled                        url=ws://[::]:8580
INFO [11-27|11:41:17.490] WebSocket enabled                        url=ws://127.0.0.1:8560
INFO [11-27|11:41:17.490] HTTP server started                      endpoint=127.0.0.1:8560 auth=true  prefix= cors=localhost vhosts=localhost
INFO [11-27|11:41:26.182] Unlocked account                         address=0x01D131ad8eF05df4643DeFf5078D98005c093e95
INFO [11-27|11:41:26.702] Unlocked antimev keystore                identity=0x01D131ad8eF05df4643DeFf5078D98005c093e95
INFO [11-27|11:41:26.702] Legacy pool tip threshold updated        tip=0
INFO [11-27|11:41:26.702] Legacy pool tip threshold updated        tip=1,000,000,000
INFO [11-27|11:41:26.702] Initializing BFT consensus               account=0x01D131ad8eF05df4643DeFf5078D98005c093e95
INFO [11-27|11:41:26.702] Snap sync complete, auto disabling
INFO [11-27|11:41:26.702] Fetching latest sealing proposal         "desired number"=1
INFO [11-27|11:41:26.703] Commit new sealing work                  number=1 sealhash=77327d..7d99aa "parent hash"=0926ce..ebed8c etherbase=0x1212000000000000000000000000000000000003 txs=0 gas=0 fees=0 elapsed="866.82µs"
INFO [11-27|11:41:27.703] Sealing proposal updated                 number=1 sealhash=77327d..7d99aa "parent hash"=0926ce..ebed8c txs=0
INFO [11-27|11:41:27.703] Starting dBFT engine                     "last height"=0 "last timestamp"=0
2024-11-27T11:41:27.703+0800	INFO	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:112	initializing dbft	{"height": 1, "view": 0, "index": -1, "role": "WatchOnly"}
INFO [11-27|11:41:27.703] dBFT event loop started
2024-11-27T11:41:27.703+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareRequest", "from": 1, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.704+0800	INFO	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:325	received PrepareRequest	{"validator": 1, "tx": 0}
2024-11-27T11:41:27.704+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareRequest", "from": 1, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:294	ignoring PrepareRequest	{"sor": true, "viewChanging": false, "moreThanF": true}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareRequest", "from": 1, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:294	ignoring PrepareRequest	{"sor": true, "viewChanging": false, "moreThanF": true}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareRequest", "from": 1, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:294	ignoring PrepareRequest	{"sor": true, "viewChanging": false, "moreThanF": true}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "RecoveryRequest", "from": 0, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "RecoveryRequest", "from": 0, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "RecoveryRequest", "from": 0, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "RecoveryRequest", "from": 0, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "RecoveryRequest", "from": 0, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "RecoveryRequest", "from": 0, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "RecoveryMessage", "from": 1, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:628	recovery message received	{"dump": {"ValidBlockStart":0,"ValidBlockEnd":1,"Sender":"0x26f1794b81df2b832545b8b6bbca196b82e4feb1","Data":"+QSTQQEBgPkEjPhG+EQBuEF2ln7zkPsFEMWi3Adlban5dLa3Ckbl5zIwUeemV95iUEYuncm/Vc6do7nmErM5M4oqllbfoamAglIMyU0Sq8i4AMDAwKDZz1kk2D7da6ppFE0ZFKawlQJwAFHXqoYjRqknDXqTS/kEHSCAgID5BBb5AhygCSbO+oklaXD/Wu8M3Nx4bqS0VMrIav4xjNyrFPPr7YygHcxN6N7HXXqrhbVntszUGtMSRRuUinQT8KFC/UDUk0eUEhIAAAAAAAAAAAAAAAAAAAAAAAOge3lo2CVG733Q6fyYb5zkTObpjlBXS9gfBCIYRX0EMOegVugfFxvMVab/g0XmksD4bltI4BuZbK3AAWIvteNjtCGgVugfFxvMVab/g0XmksD4bltI4BuZbK3AAWIvteNjtCG5AQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgGEAcnDgICEZ0aU5QCgPki8jy61AiZ7Hq5r3tLs4xopgpzflK+0TzdohwJkPkOIAAAAAAAAAAGFBKgXyACgVugfFxvMVab/g0XmksD4bltI4BuZbK3AAWIvteNjtCHAoAPpt4DmHGlbdw7pplRcjvN16uFKXLEpiYnhkooT2yNjuQHSAAtRNp0C5H7j8UM5G4N6oIwxqqGbJvF5S4HfK4MlRbi2u8oZa4Lk/rF09O/7C1OLrscDNGsDttkpL1OkzZEK0WQbcSXv90aszcofERSLIvRyxRlkATrLxrJx/uyw/r2eegEgKTDFu9llJUa8lr497JejjuM194c9+v718lCvFN9z+YPKq3sfUAIYnELgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==","Witness":"IGHnvwSYOpYzT7CIkckpqwagNT34ChBogpt+0v98U11z16b6PPU6NM0d1s8bF4G1sSPdb17Yvpl+WMfvGUw4LwE=","Type":65,"BlockIndex":1,"ValidatorIndex":1,"ViewNumber":0}}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 1, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:445	ignoring PrepareResponse from primary node	{"from": 1}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:642	recovering finished cv=0/7 preq=0/0 presp=1/7 pco=0/7 co=0/7
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "RecoveryMessage", "from": 1, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:628	recovery message received	{"dump": {"ValidBlockStart":0,"ValidBlockEnd":1,"Sender":"0x26f1794b81df2b832545b8b6bbca196b82e4feb1","Data":"+QSTQQEBgPkEjPhG+EQBuEF2ln7zkPsFEMWi3Adlban5dLa3Ckbl5zIwUeemV95iUEYuncm/Vc6do7nmErM5M4oqllbfoamAglIMyU0Sq8i4AMDAwKDZz1kk2D7da6ppFE0ZFKawlQJwAFHXqoYjRqknDXqTS/kEHSCAgID5BBb5AhygCSbO+oklaXD/Wu8M3Nx4bqS0VMrIav4xjNyrFPPr7YygHcxN6N7HXXqrhbVntszUGtMSRRuUinQT8KFC/UDUk0eUEhIAAAAAAAAAAAAAAAAAAAAAAAOge3lo2CVG733Q6fyYb5zkTObpjlBXS9gfBCIYRX0EMOegVugfFxvMVab/g0XmksD4bltI4BuZbK3AAWIvteNjtCGgVugfFxvMVab/g0XmksD4bltI4BuZbK3AAWIvteNjtCG5AQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgGEAcnDgICEZ0aU5QCgPki8jy61AiZ7Hq5r3tLs4xopgpzflK+0TzdohwJkPkOIAAAAAAAAAAGFBKgXyACgVugfFxvMVab/g0XmksD4bltI4BuZbK3AAWIvteNjtCHAoAPpt4DmHGlbdw7pplRcjvN16uFKXLEpiYnhkooT2yNjuQHSAAtRNp0C5H7j8UM5G4N6oIwxqqGbJvF5S4HfK4MlRbi2u8oZa4Lk/rF09O/7C1OLrscDNGsDttkpL1OkzZEK0WQbcSXv90aszcofERSLIvRyxRlkATrLxrJx/uyw/r2eegEgKTDFu9llJUa8lr497JejjuM194c9+v718lCvFN9z+YPKq3sfUAIYnELgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==","Witness":"IGHnvwSYOpYzT7CIkckpqwagNT34ChBogpt+0v98U11z16b6PPU6NM0d1s8bF4G1sSPdb17Yvpl+WMfvGUw4LwE=","Type":65,"BlockIndex":1,"ValidatorIndex":1,"ViewNumber":0}}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 1, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:445	ignoring PrepareResponse from primary node	{"from": 1}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:642	recovering finished cv=0/7 preq=0/0 presp=1/7 pco=0/7 co=0/7
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "RecoveryRequest", "from": 0, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 0, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	INFO	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:464	received PrepareResponse	{"validator": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 0, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:452	ignoring PrepareResponse	{"dup": true, "sor": true, "viewChanging": false, "moreThanF": true}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 0, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.705+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:452	ignoring PrepareResponse	{"dup": true, "sor": true, "viewChanging": false, "moreThanF": true}
2024-11-27T11:41:27.773+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 2, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.773+0800	INFO	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:464	received PrepareResponse	{"validator": 2}
2024-11-27T11:41:27.779+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 2, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.779+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:452	ignoring PrepareResponse	{"dup": true, "sor": true, "viewChanging": false, "moreThanF": true}
2024-11-27T11:41:27.779+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "RecoveryMessage", "from": 2, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.779+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:628	recovery message received	{"dump": {"ValidBlockStart":0,"ValidBlockEnd":1,"Sender":"0x74f4effb0b538baec703346b03b6d9292f53a4cd","Data":"+QTZQQECgPkE0viM+EQBuEF2ln7zkPsFEMWi3Adlban5dLa3Ckbl5zIwUeemV95iUEYuncm/Vc6do7nmErM5M4oqllbfoamAglIMyU0Sq8i4APhEArhBfacW28dqBdlgmCvLIlQh0LQ8tG6WDtiKf+nVi8DTWWls4ZWlNGKmZ6Be5p8TCwisAMbQLXeDgM7Uq7S0u42/3ADAwMCg2c9ZJNg+3WuqaRRNGRSmsJUCcABR16qGI0apJw16k0v5BB0ggICA+QQW+QIcoAkmzvqJJWlw/1rvDNzceG6ktFTKyGr+MYzcqxTz6+2MoB3MTejex116q4W1Z7bM1BrTEkUblIp0E/ChQv1A1JNHlBISAAAAAAAAAAAAAAAAAAAAAAADoHt5aNglRu990On8mG+c5Ezm6Y5QV0vYHwQiGEV9BDDnoFboHxcbzFWm/4NF5pLA+G5bSOAbmWytwAFiL7XjY7QhoFboHxcbzFWm/4NF5pLA+G5bSOAbmWytwAFiL7XjY7QhuQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIBhAHJw4CAhGdGlOUAoD5IvI8utQImex6ua97S7OMaKYKc35SvtE83aIcCZD5DiAAAAAAAAAABhQSoF8gAoFboHxcbzFWm/4NF5pLA+G5bSOAbmWytwAFiL7XjY7QhwKAD6beA5hxpW3cO6aZUXI7zderhSlyxKYmJ4ZKKE9sjY7kB0gALUTadAuR+4/FDORuDeqCMMaqhmybxeUuB3yuDJUW4trvKGWuC5P6xdPTv+wtTi67HAzRrA7bZKS9TpM2RCtFkG3El7/dGrM3KHxEUiyL0csUZZAE6y8aycf7ssP69nnoBICkwxbvZZSVGvJa+PeyXo47jNfeHPfr+9fJQrxTfc/mDyqt7H1ACGJxC4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=","Witness":"rt1TS/TazPVNdU9T+CuXtlpArc3ciBgiYfECdav8ITxCUM3OHPAYapR0v9e9dHqOetE+gS/zTPBHzbGjtmbyqwE=","Type":65,"BlockIndex":1,"ValidatorIndex":2,"ViewNumber":0}}
2024-11-27T11:41:27.779+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 1, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.779+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:445	ignoring PrepareResponse from primary node	{"from": 1}
2024-11-27T11:41:27.779+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 2, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.779+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:452	ignoring PrepareResponse	{"dup": true, "sor": true, "viewChanging": false, "moreThanF": true}
2024-11-27T11:41:27.779+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:642	recovering finished cv=0/7 preq=0/0 presp=2/7 pco=0/7 co=0/7
2024-11-27T11:41:27.779+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 2, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.779+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:452	ignoring PrepareResponse	{"dup": true, "sor": true, "viewChanging": false, "moreThanF": true}
2024-11-27T11:41:27.785+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 2, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.785+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:452	ignoring PrepareResponse	{"dup": true, "sor": true, "viewChanging": false, "moreThanF": true}
2024-11-27T11:41:27.790+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 2, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.791+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:452	ignoring PrepareResponse	{"dup": true, "sor": true, "viewChanging": false, "moreThanF": true}
2024-11-27T11:41:27.794+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "RecoveryMessage", "from": 2, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.794+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:628	recovery message received	{"dump": {"ValidBlockStart":0,"ValidBlockEnd":1,"Sender":"0x74f4effb0b538baec703346b03b6d9292f53a4cd","Data":"+QTZQQECgPkE0viM+EQBuEF2ln7zkPsFEMWi3Adlban5dLa3Ckbl5zIwUeemV95iUEYuncm/Vc6do7nmErM5M4oqllbfoamAglIMyU0Sq8i4APhEArhBfacW28dqBdlgmCvLIlQh0LQ8tG6WDtiKf+nVi8DTWWls4ZWlNGKmZ6Be5p8TCwisAMbQLXeDgM7Uq7S0u42/3ADAwMCg2c9ZJNg+3WuqaRRNGRSmsJUCcABR16qGI0apJw16k0v5BB0ggICA+QQW+QIcoAkmzvqJJWlw/1rvDNzceG6ktFTKyGr+MYzcqxTz6+2MoB3MTejex116q4W1Z7bM1BrTEkUblIp0E/ChQv1A1JNHlBISAAAAAAAAAAAAAAAAAAAAAAADoHt5aNglRu990On8mG+c5Ezm6Y5QV0vYHwQiGEV9BDDnoFboHxcbzFWm/4NF5pLA+G5bSOAbmWytwAFiL7XjY7QhoFboHxcbzFWm/4NF5pLA+G5bSOAbmWytwAFiL7XjY7QhuQEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIBhAHJw4CAhGdGlOUAoD5IvI8utQImex6ua97S7OMaKYKc35SvtE83aIcCZD5DiAAAAAAAAAABhQSoF8gAoFboHxcbzFWm/4NF5pLA+G5bSOAbmWytwAFiL7XjY7QhwKAD6beA5hxpW3cO6aZUXI7zderhSlyxKYmJ4ZKKE9sjY7kB0gALUTadAuR+4/FDORuDeqCMMaqhmybxeUuB3yuDJUW4trvKGWuC5P6xdPTv+wtTi67HAzRrA7bZKS9TpM2RCtFkG3El7/dGrM3KHxEUiyL0csUZZAE6y8aycf7ssP69nnoBICkwxbvZZSVGvJa+PeyXo47jNfeHPfr+9fJQrxTfc/mDyqt7H1ACGJxC4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=","Witness":"rt1TS/TazPVNdU9T+CuXtlpArc3ciBgiYfECdav8ITxCUM3OHPAYapR0v9e9dHqOetE+gS/zTPBHzbGjtmbyqwE=","Type":65,"BlockIndex":1,"ValidatorIndex":2,"ViewNumber":0}}
2024-11-27T11:41:27.794+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 1, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.794+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:445	ignoring PrepareResponse from primary node	{"from": 1}
2024-11-27T11:41:27.794+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 2, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:27.794+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:452	ignoring PrepareResponse	{"dup": true, "sor": true, "viewChanging": false, "moreThanF": true}
2024-11-27T11:41:27.794+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:642	recovering finished cv=0/7 preq=0/0 presp=2/7 pco=0/7 co=0/7
2024-11-27T11:41:28.413+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 5, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:28.413+0800	INFO	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:464	received PrepareResponse	{"validator": 5}
2024-11-27T11:41:28.417+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "PrepareResponse", "from": 5, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
2024-11-27T11:41:28.417+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:452	ignoring PrepareResponse	{"dup": true, "sor": true, "viewChanging": false, "moreThanF": true}
2024-11-27T11:41:28.433+0800	DEBUG	dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:231	received message	{"type": "RecoveryRequest", "from": 3, "height": 1, "view": 0, "my_height": 1, "my_view": 0}
panic: runtime error: index out of range [-1]

goroutine 4484 [running]:
github.com/ethereum/go-ethereum/consensus/dbft.(*DBFT).newPayload(...)
	github.com/ethereum/go-ethereum/consensus/dbft/dbft.go:1913
github.com/nspcc-dev/dbft.(*Context[...]).makeRecoveryMessage(0x1fc2820)
	github.com/nspcc-dev/dbft@v0.3.1-0.20241030192742-82ad77871e88/send.go:199 +0x18c
github.com/nspcc-dev/dbft.(*DBFT[...]).sendRecoveryMessage(0x1fc5f60)
	github.com/nspcc-dev/dbft@v0.3.1-0.20241030192742-82ad77871e88/send.go:203 +0x29
github.com/nspcc-dev/dbft.(*DBFT[...]).onRecoveryRequest(0x1fc5f60, {0x1fab058, 0xc0001abc00})
	github.com/nspcc-dev/dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:624 +0x108
github.com/nspcc-dev/dbft.(*DBFT[...]).OnReceive(0x1fc5f60, {0x1fab058, 0xc0001abc00})
	github.com/nspcc-dev/dbft@v0.3.1-0.20241030192742-82ad77871e88/dbft.go:282 +0x1075
github.com/ethereum/go-ethereum/consensus/dbft.(*DBFT).eventLoop(0xc0004ce960)
	github.com/ethereum/go-ethereum/consensus/dbft/dbft.go:1707 +0xae5
created by github.com/ethereum/go-ethereum/consensus/dbft.(*DBFT).Start in goroutine 4431
	github.com/ethereum/go-ethereum/consensus/dbft/dbft.go:1551 +0x476
```



#### Steps to reproduce the behaviour
1. create a new account, for example node9
2. change the Makefile to start the node in miner mode together with other nodes
[Makefile.txt](https://github.com/user-attachments/files/17928728/Makefile.txt)
3. run the command 
`make privnet_start_seven`


@AnnaShaleva @txhsl @chenquanyu

--- a/go.mod
+++ b/go.mod
@@ -51,7 +51,7 @@ require (
 	github.com/mattn/go-colorable v0.1.13
 	github.com/mattn/go-isatty v0.0.19
 	github.com/naoina/toml v0.1.2-0.20170918210437-9fafd6967416
-	github.com/nspcc-dev/dbft v0.3.1-0.20241030192742-82ad77871e88
+	github.com/nspcc-dev/dbft v0.3.1-0.20241128121733-94bc38235e44
 	github.com/olekukonko/tablewriter v0.0.5
 	github.com/peterh/liner v1.1.1-0.20190123174540-a2c9a5303de7
 	github.com/protolambda/bls12-381-util v0.0.0-20220416220906-d8552aa452c7

